app/models/vaccination_record.rb

```diff
@@ -18,4 +20,18 @@
 #
 class VaccinationRecord < ApplicationRecord
   belongs_to :patient_session
+
+  enum :site, %i[left_arm right_arm other]
+
+  validates :administered, inclusion: [true, false]
+  validates :site,
+            presence: true,
+            inclusion: {
+              in: sites.keys
+            },
+            if: -> { administered }
+
+  def self.site_options
```

Comment:
I'd be tempted to put this into a helper, since I think this is only useful in for views.